src/Driver.cpp

```diff
@@ -194,6 +194,18 @@ void Driver::openURI(std::string const& uri)
     { // UDP udp://hostname:remoteport
         if (marker == string::npos)
             throw std::runtime_error("missing port specification in udp:// URI");
+
+        string::size_type second_marker = device.find_last_of(":");
+        int extended_info = 0;
```

Comment:
Declare local variables in the deepest scope possible, and where they are used first.